desc/grid.py

```diff
@@ -139,13 +139,30 @@ def _scale_weights(self, dtheta_scale):
         _, inverse, counts = np.unique(
             nodes, axis=0, return_inverse=True, return_counts=True
         )
-        self._spacing /= np.tile(np.atleast_2d(counts[inverse]).T, 3) ** (1 / 3)
-        # assign weights pretending _enforce_symmetry didn't change theta spacing
+        duplicates = np.tile(np.atleast_2d(counts[inverse]).T, 3)
         temp_spacing = np.copy(self.spacing)
+        temp_spacing /= duplicates ** (1 / 3)
+        # assign weights pretending _enforce_symmetry didn't change theta spacing
         temp_spacing[:, 1] /= dtheta_scale
+        # scale weights sum to full volume
         temp_spacing *= (4 * np.pi**2 / temp_spacing.prod(axis=1).sum()) ** (1 / 3)
         self._weights = temp_spacing.prod(axis=1)
-        self._spacing *= (4 * np.pi**2 / self.spacing.prod(axis=1).sum()) ** (1 / 3)
+
+        # Spacing is rescaled below. Note this doesn't work in full generality.
+        # (Duplicate nodes should typically be deleted rather than rescaled).
+
+        # Spacing is the differential element used for integration over surfaces.
+        # For this, 2 columns of the matrix are used.
+        # The reduction of weight on duplicate nodes should be accounted for
+        # by the 2 columns of spacing which span the surface.
+        # Note we rescale 3 columns by the factor that 'should' rescale 2 columns,
+        # so the grid spacing is valid for integrals over all surface labels.
+        self._spacing /= duplicates ** (1 / 2)
+        # Multiply by duplicates^(1/6) to account for the extra division by
+        # duplicates^(1/2) in one of the columns above. 1/2 = (1/6)^(3 columns).
```

Comment:
I'll probably delete this line and mention that we're trying to preserve double product of each pair of spacing columns.